src/model/base.ts

```diff
@@ -1,12 +1,82 @@
-import { Tensor, tensor, RecursiveArray } from '@tensorflow/tfjs-core';
+import { Tensor, RecursiveArray, tensor, tidy, dispose } from '@tensorflow/tfjs-core';
 import { checkArray } from '../utils/validation';
 import { OneHotEncoder } from '../preprocess';
 import { OneHotDropTypes } from '../preprocess/encoder';
 
+export type FeatureInputType = Tensor | RecursiveArray<number>;
+export type LabelInputType = Tensor | RecursiveArray<number>;
+
 export type ClassMap = {
   [ key: string ]: number
 }
 
+export abstract class BaseEstimator {
+  public estimatorType: string;
+  public nFeature: number;
+
+  /**
+   * Check if input feature match the required feature size. If reset is true,
+   * reset nFeature = x.shape[1]
+   *
+   * @param x Input feature of shape (nSample, nFeatures).
+   * @param reset if true, the `nFeatures` attribute is set to `x.shape[1]`.
+   */
+  public checkAndSetNFeatures(x: Tensor, reset: boolean): void {
+    if (x?.shape?.length !== 2) {
+      throw new TypeError('Input should be 2D tensor');
```

Comment:
```suggestion
      throw new TypeError('input should be 2D tensor');
```
I just found other messages are in lower-case for the first char.